Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// @file GaussianSplatSpaceComponent.h
+/// @brief Definitions and support for Gaussian Splats.
+
+#pragma once
+
+#include "CSP/CSPCommon.h"
+#include "CSP/Common/String.h"
+#include "CSP/Multiplayer/ComponentBase.h"
+#include "CSP/Multiplayer/Components/Interfaces/IExternalResourceComponent.h"
+#include "CSP/Multiplayer/Components/Interfaces/IShadowCasterComponent.h"
+#include "CSP/Multiplayer/Components/Interfaces/IThirdPartyComponentRef.h"
+#include "CSP/Multiplayer/Components/Interfaces/ITransformComponent.h"
+#include "CSP/Multiplayer/Components/Interfaces/IVisibleComponent.h"
+
+
+namespace csp::multiplayer
+{
+
+/// @brief Enumerates the list of properties that can be replicated for a Gaussian Splat component.
+enum class GaussianSplatPropertyKeys
+{
+	Name = 0,
+	ExternalResourceAssetId,
+	ExternalResourceAssetCollectionId,
+	Position,
+	Rotation,
+	Scale,
+	IsVisible,
+	IsARVisible,
+	ThirdPartyComponentRef,
```

Comment:
What value does this have in the context of gsplats?

Reply:
What ThirdPartyComponentRef?

Reply:
Yeah

Reply:
Removed.